bin/filter_sce_rds.R

```diff
@@ -99,13 +126,59 @@ alt_names <- altExpNames(filtered_sce)
 
 for (alt in alt_names) {
   # remove old row data from unfiltered
-  drop_cols = colnames(rowData(altExp(filtered_sce, alt))) %in% c('mean', 'detected')
+  drop_cols <- colnames(rowData(altExp(filtered_sce, alt))) %in% c('mean', 'detected')
   rowData(altExp(filtered_sce, alt)) <- rowData(altExp(filtered_sce, alt))[!drop_cols]
 
   # add alt experiment features stats for filtered data
   altExp(filtered_sce, alt) <- scuttle::addPerFeatureQCMetrics(altExp(filtered_sce, alt))
 }
 
+
+# calculate filtering QC from ADTs, if present
+if (!is.null(ambient_profile)) {
+  
+  # Create data frame of ADTs and their target types
+  # If `target_type` column is not present, assume all ADTs are targets
+  adt_barcode_df <- readr::read_tsv(
+    opt$adt_barcode_file, 
+    # if only 2 columns exist, only the first two col_names will be used
+    col_names = c("name", "barcode", "target_type")
+  ) 
+  if (!"target_type" %in% names(adt_barcode_df)) {
+    adt_barcode_df$target_type <- "target"
+  } 
+ if( !all.equal(sort(adt_barcode_df$names), sort(rownames(altExp(filtered_sce, adt_exp)) ){
+   stop("Mismatch between provided ADT barcode file and ADTs in SCE.")
+ }
```

Comment:
My suggestion, my bad spacing
```suggestion
  # Check that barcodes match
  if( !all.equal(sort(adt_barcode_df$names), sort(rownames(altExp(filtered_sce, adt_exp)) ){
    stop("Mismatch between provided ADT barcode file and ADTs in SCE.")
  }

```

Reply:
beat you to it.. https://github.com/AlexsLemonade/scpca-nf/pull/312/commits/40c5914367760d93863d2d8a3aaec017353792ff

Reply:
honestly without the rainbow parentheses this seems ~impossible to get right!

Reply:
And apparenlty parens (🤦🏼 )